test-alerts.cpp

```diff
@@ -5,4 +5,37 @@
 
 TEST_CASE("infers the breach according to limits") {
   REQUIRE(inferBreach(12, 20, 30) == TOO_LOW);
+  REQUIRE(inferBreach(40, 20, 30) == TOO_HIGH);
+  REQUIRE(inferBreach(25, 20, 30) == NORMAL);
 }
+
+TEST_CASE("classify temp according to limits") {
+    REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, -1)== TOO_LOW); 
+    REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING, -10)== TOO_LOW); 
+    REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, -100)== TOO_LOW); 
+    REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, 36)== TOO_HIGH); 
+    REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING, 41)== TOO_HIGH); 
+    REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, 46)== TOO_HIGH); 
+    REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, 33)== NORMAL); 
+    REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING, 39)== NORMAL); 
+    REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, 44)== NORMAL); 
+}
+
+TEST_CASE("check and alert based on temp range") {
+  BatteryCharacter batteryChar ={PASSIVE_COOLING,{0}};
+  
+  batteryChar.coolingType = PASSIVE_COOLING;
+
+ checkAndAlert(TO_CONTROLLER,batteryChar, -1); 
+ checkAndAlert(TO_EMAIL,batteryChar ,10); 
+ checkAndAlert(NO_ALERT,batteryChar ,34); 
+}
+
+TEST_CASE("check breach type") {
+
+   sendToEmail(TOO_LOW); 
```

Comment:
These tests call the function, but don't test the expected functionality. Consequence is that the testing would have to be done with real controllers and someone's email id.
Try dependency injection to make an interface and implement the stub to remember the email sent in a data member. You can assert on it then.